pkg/kubectl/cmd/util/factory.go

```diff
@@ -227,6 +235,20 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 
 	clients := NewClientCache(clientConfig)
 
+	// We do this to delay loading a config until after flags have been
+	// parsed
+	var dynamicClientPool dynamic.ClientPool
+	var dynamicClientPoolOnce sync.Once
+	getDynamicClientPool := func() dynamic.ClientPool {
+		dynamicClientPoolOnce.Do(func() {
+			cfg, err := clientConfig.ClientConfig()
+			CheckErr(err)
+
+			dynamicClientPool = dynamic.NewClientPool(cfg, dynamic.LegacyAPIPathResolverFunc)
+		})
+		return dynamicClientPool
+	}
```

Comment:
Hey @smarterclayton I changed this to delay the client loading for the client pool. Turns out my e2e failures weren't a flake, but a result of kubectl not picking up the config location from the command line. Hopefully with this delay, it works. Could you re-review just this portion?